target_chains/sui/contracts/sources/pyth.move

```diff
@@ -85,8 +92,47 @@ module pyth::pyth {
         sources
     }
 
-    /// Create and share new price feed objects if they don't already exist.
-    public fun create_price_feeds(
+    /// Create and share new price feed objects if they don't already exist using accumulator message.
+    public fun create_price_feeds_using_accumulator(
+        pyth_state: &mut PythState,
+        accumulator_message: vector<u8>,
+        vaa: VAA, // the verified version of the vaa bytes encoded within the accumulator_message
+        clock: &Clock,
+        ctx: &mut TxContext
+    ){
+        // This capability ensures that the current build version is used.
+        let latest_only = state::assert_latest_only(pyth_state);
+
+        // Check that the VAA is from a valid data source (emitter)
+        assert!(
+            state::is_valid_data_source(
+                pyth_state,
+                data_source::new(
+                    (vaa::emitter_chain(&vaa) as u64),
+                    vaa::emitter_address(&vaa))
+                ),
+            E_INVALID_DATA_SOURCE
+        );
+
+        // decode the price info updates from the VAA payload (first check if it is an accumulator or batch price update)
+        let accumulator_message_cursor = cursor::new(accumulator_message);
+        let header = deserialize::deserialize_u32(&mut accumulator_message_cursor);
+
+        if ((header as u64) != PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
+            abort E_INVALID_ACCUMULATOR_HEADER
+        };
```

Comment:
should this check go in parse_and_verify_accumulator_message? that seems safer and would reduce code duplication -- I see a very similar if block below.